widget/entry-with-completion.go

```diff
@@ -0,0 +1,176 @@
+package widget
+
+import (
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/theme"
+	"fyne.io/fyne/v2/widget"
+)
+
+// CompletionEntry is an Entry with options displayed in a PopUpMenu.
+type CompletionEntry struct {
+	widget.Entry
+	popupMenu     *widget.PopUp
+	navigableList *navigableList
+	Options       []string
+	pause         bool
+}
+
+// NewCompletionEntry creates a new CompletionEntry which creates a popup menu that responds to keystrokes to navigate through the items without losing the editing ability of the text input.
+func NewCompletionEntry(options []string) *CompletionEntry {
+	c := &CompletionEntry{Options: options}
+	c.ExtendBaseWidget(c)
+	return c
+}
+
+// ShowCompletion displays the completion menu
+func (c *CompletionEntry) ShowCompletion() {
+	if c.pause {
+		return
+	}
+
+	if c.navigableList == nil {
+		c.navigableList = newNavigableList(c.Options, &c.Entry, c.setTextFromMenu, c.HideCompletion)
+	}
+	holder := fyne.CurrentApp().Driver().CanvasForObject(c)
+
+	if c.popupMenu == nil {
+		c.popupMenu = widget.NewPopUp(c.navigableList, holder)
+	}
+	c.popupMenu.Resize(c.maxSize())
+	c.popupMenu.ShowAtPosition(c.popUpPos())
+	holder.Focus(c.navigableList)
+}
+
+// HideCompletion hides the completion menu.
+func (c *CompletionEntry) HideCompletion() {
+	if c.popupMenu != nil {
+		c.popupMenu.Hide()
+	}
+}
+
+// SetOptions set the completion list with itemList and update the view.
+func (c *CompletionEntry) SetOptions(itemList []string) {
+	c.Options = itemList
+	c.Refresh()
+}
+
+// Refresh the list to update the options to display.
+func (c *CompletionEntry) Refresh() {
+	c.Entry.Refresh()
+	if c.navigableList != nil {
+		c.navigableList.SetOptions(c.Options)
+	}
+}
+
+// calculate where the popup should appear
+func (c *CompletionEntry) popUpPos() fyne.Position {
+	entryPos := fyne.CurrentApp().Driver().AbsolutePositionForObject(c)
+	return entryPos.Add(fyne.NewPos(0, c.Size().Height))
+}
+
+// calculate the max size to make the popup to cover everything below the entry
+func (c *CompletionEntry) maxSize() fyne.Size {
+	cnv := fyne.CurrentApp().Driver().CanvasForObject(c)
+	return fyne.NewSize(c.Size().Width, cnv.Size().Height-c.Position().Y-c.Size().Height-theme.InputBorderSize()-theme.Padding())
```

Comment:
This should not draw over the whole canvas.
It should limit the height to the size of the list - otherwise it covers the entire area of the app below the entry (often with a lot of empty space).

Reply:
It's resolved in https://github.com/fyne-io/fyne-x/pull/11/commits/d82cdaef8120e90209180b7fede7d19e5ca14279 if I did things well ;)